### PR TITLE
update DataLoader doc

### PR DESCRIPTION
**Pull request by heavengate**
### PR types
Others

### PR changes
Docs

### Describe
update doc for `paddle.io.DataLoader`
- support `map-stype` and `iterable-style`
- remove `fluid` & `LoDTensor`
- add `from_generator` `from_dataset` deprecated doc

dataloader1
dataloader2
dataloader3
dataloader4
屏幕快照 2020-10-29 下午9 58 29

### CODE REVIEW

File: python/paddle/fluid/reader.py [hunk 1]
```diff
@@ -251,63 +256,62 @@ def __len__(self):
...
 
+            # get places
+            place = paddle.CUDAPlace(0) if USE_GPU else paddle.CPUPlace()
+
             def simple_net(image, label):
-                fc_tmp = fluid.layers.fc(image, size=CLASS_NUM, act='softmax')
-                cross_entropy = fluid.layers.softmax_with_cross_entropy(image, label)
-                loss = fluid.layers.reduce_mean(cross_entropy)
-                sgd = fluid.optimizer.SGD(learning_rate=1e-3)
+                fc_tmp = paddle.fluid.layers.fc(image, size=CLASS_NUM, act='softmax')
```
Comment (saxon-zh) [inline python/paddle/fluid/reader.py]: 还带有fluid

File: python/paddle/fluid/reader.py [hunk 2]
```diff
@@ -439,6 +443,10 @@ def from_generator(feed_list=None,
                        use_multiprocess=False,
                        drop_last=True):
         """
+        .. note::
```
Comment (saxon-zh) [inline python/paddle/fluid/reader.py]: 换成warning类型吧

File: python/paddle/fluid/reader.py (hunk 1 above)
Comment (heavengate) [reply python/paddle/fluid/reader.py]: Done, thanks!

File: python/paddle/fluid/reader.py (hunk 2 above)
Comment (heavengate) [reply python/paddle/fluid/reader.py]: Done, thanks!

**Review by saxon-zh — APPROVED** [review]
LGTM

**Pull request merged**